src/Symfony/Component/Validator/Constraints/UrlValidator.php

```diff
@@ -21,34 +21,36 @@
  */
 class UrlValidator extends ConstraintValidator
 {
-    public const PATTERN = '~^
+    public const PATTERN = <<<'REGEX'
+        {^
             (%s)://                                 # protocol
-            (((?:[\_\.\pL\pN-]|%%[0-9A-Fa-f]{2})+:)?((?:[\_\.\pL\pN-]|%%[0-9A-Fa-f]{2})+)@)?  # basic auth
+            ((?:[\pL\pN\-._~!$&'()*+,;=]|%%[0-9A-Fa-f]{2})++(?::(?:[:\pL\pN\-._~!$&'()*+,;=]|%%[0-9A-Fa-f]{2})*+)?@)?  # basic auth
```

Comment:
The RFC allows empty user part, so technically we could go with just the following.
But this makes tests fail as we consider userinfo with no username as invalid (`http://:pwd@example.com`)

```suggestion
            ((?:[:\pL\pN\-._~!$&'()*+,;=]|%%[0-9A-Fa-f]{2})++)@)?  # basic auth
```

(support for basic auth was added 11 years ago in #11601)